src/main/java/com/github/kay9/dragonmounts/dragon/ai/DragonFollowOwnerGoal.java

```diff
@@ -0,0 +1,179 @@
+package com.github.kay9.dragonmounts.dragon.ai;
+
+import com.github.kay9.dragonmounts.dragon.TameableDragon;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.ai.goal.Goal;
+import net.minecraft.world.entity.ai.navigation.FlyingPathNavigation;
+import net.minecraft.world.entity.ai.navigation.GroundPathNavigation;
+import net.minecraft.world.level.LevelReader;
+import net.minecraft.world.level.block.LeavesBlock;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.pathfinder.BlockPathTypes;
+import net.minecraft.world.level.pathfinder.WalkNodeEvaluator;
+
+import java.util.EnumSet;
+
+/**
+ * Goal for dragon to follow its owner.
+ * <p></p>
+ * Mostly copied from <code>FollowOwnerGoal</code>, but with some modifications to fix an issue.
+ * Also allows dragon to tp to owner in the air so they don't get stuck until the owner lands.
+ *
+ * @author AnimalsWritingCode
+ *
+ * @see net.minecraft.world.entity.ai.goal.FollowOwnerGoal
+ */
+public class DragonFollowOwnerGoal extends Goal {
```

Comment:
formatting; I don't blame you for missing it though, as it seems like a modified copy+paste.